tests/neptune/new/internal/utils/test_deprecation.py

```diff
@@ -13,51 +13,65 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-import unittest
-from unittest import mock
-from unittest.mock import call
+import pytest
 
 from neptune.new.exceptions import NeptuneParametersCollision
-from neptune.new.internal.utils.deprecation import deprecated_parameter
+from neptune.new.internal.utils.deprecation import deprecated, deprecated_parameter
 
 
 @deprecated_parameter(deprecated_kwarg_name="deprecated_param", required_kwarg_name="new_param")
-def fun(*, new_param):
+def fun_with_deprecated_param(*, new_param):
     return new_param
 
 
-class TestDeprecatedParameter(unittest.TestCase):
-    @mock.patch("neptune.new.internal.utils.deprecation.logger")
-    def test_deprecated_not_used(self, logger_mock):
-        self.assertEqual(42, fun(new_param=42))
-        self.assertFalse(logger_mock.called)
+@deprecated()
+def deprecated_func():
+    pass
 
-    @mock.patch("neptune.new.internal.utils.deprecation.logger")
-    def test_deprecated_replaced(self, logger_mock):
+
+@deprecated(alternative="non_deprecated_func")
+def deprecated_func_with_alternative():
+    pass
+
+
+class TestDeprecatedParameter:
+    def test_deprecated_not_used(self):
+        with pytest.warns(None):
+            fun_with_deprecated_param(new_param=42)
+
+    def test_deprecated_replaced(self):
         # pylint: disable=unexpected-keyword-arg
         # pylint: disable=missing-kwoa
-        self.assertEqual(42, fun(deprecated_param=42))
-        self.assertEqual(1, logger_mock.warning.call_count)
-        self.assertEqual(
-            call(
-                "parameter `%s` is deprecated, use `%s` instead."
-                " We'll end support of it in `neptune-client==1.0.0`.",
-                "deprecated_param",
-                "new_param",
-            ),
-            logger_mock.warning.call_args,
-        )
+        with pytest.deprecated_call(
+            match="Parameter `deprecated_param` is deprecated, use `new_param` instead. We'll end support of it in "
+            "`neptune-client==1.0.0`."
+        ):
+            value = fun_with_deprecated_param(deprecated_param=42)
+            assert value == 42
 
     def test_conflict(self):
-        with self.assertRaises(NeptuneParametersCollision):
-            # pylint: disable=unexpected-keyword-arg
-            fun(new_param=42, deprecated_param=42)
+        # pylint: disable=unexpected-keyword-arg
+        with pytest.raises(NeptuneParametersCollision):
+            fun_with_deprecated_param(new_param=42, deprecated_param=42)
 
     def test_passing_deprecated_parameter_as_none(self):
         # pylint: disable=unexpected-keyword-arg
         # pylint: disable=missing-kwoa
-        self.assertIsNone(fun(deprecated_param=None))
```

Comment:
Why did we removed this line?

Reply:
Hmm, probably missed that and it should be replaced with `assert value is None`

Reply:
Done